src/components/compress/index.tsx

```diff
@@ -38,6 +38,51 @@ import { ExpandIcon, CopyAcrossIconProps } from '../../lib/icons';
 import SnackBarElement from '../../lib/SnackBar';
 import { InputProcessorState, defaultInputProcessorState } from '../../codecs/input-processors';
 
+// Safari and Edge don't quite support extending Event, this works around it.
+function fixExtendedEvent(instance: Event, type: Function) {
+  if (!(instance instanceof type)) {
+    Object.setPrototypeOf(instance, type.prototype);
+  }
+}
+export enum SquooshStartEventType {
+  START = 'squoosh:start',
+}
+export class SquooshStartEvent extends Event {
+  constructor(init?: EventInit) {
+    super(SquooshStartEventType.START, init);
+    fixExtendedEvent(this, SquooshStartEvent);
+  }
+}
+
+export enum SquooshSideEventType {
+  DONE = 'squoosh:done',
+  ABORT = 'squoosh:abort',
+  ERROR = 'squoosh:error',
+}
```

Comment:
Do we need the extra enum features here, or will `const enum` do?

```ts
enum SquooshSideEventType {
  DONE = 'squoosh:done',
  ABORT = 'squoosh:abort',
  ERROR = 'squoosh:error',
}

console.log(SquooshSideEventType.DONE);
```

becomes

```js
var SquooshSideEventType;
(function (SquooshSideEventType) {
    SquooshSideEventType["DONE"] = "squoosh:done";
    SquooshSideEventType["ABORT"] = "squoosh:abort";
    SquooshSideEventType["ERROR"] = "squoosh:error";
})(SquooshSideEventType || (SquooshSideEventType = {}));

console.log(SquooshSideEventType.DONE);
```

Whereas

```ts
const enum SquooshSideEventType {
  DONE = 'squoosh:done',
  ABORT = 'squoosh:abort',
  ERROR = 'squoosh:error',
}

console.log(SquooshSideEventType.DONE);
```

Becomes

```js
console.log("squoosh:done" /* DONE */);
```